examples/text-generation/run_generation.py

```diff
@@ -313,8 +321,164 @@ def main():
         use_lazy_mode = False
 
     import habana_frameworks.torch.hpu as torch_hpu
+    if args.dataset_name == "openorca":
+        # Benchmark over the prompts below
+        def get_ds(args):
+            ds = pd.read_pickle(args.dataset)
+            return ds
+
+
+        def get_input(ds, batch_size):
+            queries = []
+            tok_input = ds["tok_input"].tolist()
+            for start in range(0, len(ds), batch_size):
+                end = start + batch_size
+                batch = tok_input[start:end]
+                input_ids = []
+                attention_mask=[]
+                for query in batch:
+                    input_ids.append(
+                        [0] * (args.max_input_tokens - len(query)) + query)
+                    attention_mask.append([0] * (args.max_input_tokens - len(query)) + [1] * len(query))
+                queries.append({
+                    'input_ids': torch.tensor(input_ids, dtype=torch.int32),
+                    'attention_mask': torch.tensor(attention_mask, dtype=torch.int32)
+                })
+            return queries
+
+        ds = get_ds(args)
+        input_sentences = get_input(ds, args.batch_size)
+
+        def generate(input_tokens, size=None, reduce_recompile=False):
+            """Generates sequences from the input sentences and returns them."""
+
+            t0 = time.perf_counter()
+            print(f"Step4+ starting time is {t0*1000}", flush=True)
+            if size is not None:
+                input_tokens = adjust_batch(input_tokens, size)
+
+            if not reduce_recompile:
+                # Move inputs to target device(s)
+                for t in input_tokens:
+                    if torch.is_tensor(input_tokens[t]):
+                        input_tokens[t] = input_tokens[t].to(args.device)
+
+            outputs = model.generate(
+                **input_tokens,
+                generation_config=generation_config,
+                lazy_mode=use_lazy_mode,
+                hpu_graphs=args.use_hpu_graphs,
+                profiling_steps=args.profiling_steps,
+                profiling_warmup_steps=args.profiling_warmup_steps,
+            ).cpu()
+            outputs = outputs.tolist()
+            for i in range(len(outputs)):
+                outputs[i] = outputs[i][args.max_input_tokens:]
+            duration = time.perf_counter() - t0
+            print(f"Total E2E time of this batch is {duration:.3f}s", flush=True)
+            return outputs
```

Comment:
This is a LOT of code duplication, which can cause errors later due to failing to remember to propagate chaneges in both branches etc.


For example, you could reuse the older "generate" function, just by adding:
```
def generate(input_tokens=None, size=None, reduce_recompile=False):
.....
    if input_tokens is None:  # ADDING THIS
				# Tokenization
				if args.max_input_tokens > 0:
					input_tokens = tokenizer.batch_encode_plus(
						input_sentences,
						return_tensors="pt",
						padding="max_length",
						max_length=args.max_input_tokens,
						truncation=True,
					)
				else:
					input_tokens = tokenizer.batch_encode_plus(input_sentences, return_tensors="pt", padding=True)
```

And you could perform the output runcation outside generate:
```
for i in range(len(outputs)):
                outputs[i] = outputs[i][args.max_input_tokens:]
```

Reply:
please note we need to run the measurement also, we need other functionalities from this file therefore existing behaviours are retained, if any code duplication can be handled as part of later cleanup/code refactoring.